src/MaxText/kernels/megablox/ops.py

```diff
@@ -114,29 +120,47 @@ def _gmm_fwd(
           calibration_method=quantization_rule.weight_calibration_method,
           scale_dtype=jnp.float32,
       )
-
-  out = backend.gmm(
-      lhs,
-      rhs,
-      group_sizes,
-      preferred_element_type,
-      tiling,
-      group_offset,
-      existing_out,
-      transpose_rhs=transpose_rhs,
-      interpret=interpret,
-  )
+      # QAG is only supported for following conditions
+  if use_tokamax_backend:
+    if quantization_rule and quantization_rule.bwd_qtype:
+      if quantization_rule.weight_calibration_method.startswith("fixed") and isinstance(rhs, qpl.QArray):
+        rhs_qvalue = jax.lax.all_gather(rhs.qvalue, "fsdp", axis=0, tiled=True)
+        rhs = dataclasses.replace(rhs, qvalue=rhs_qvalue)
+    out = tokamax_backend.gmm(
```

Comment:
Ah... It looks like we integrated the tokamax quantization logic into megablox ops.